addons/mail/controllers/discuss.py

```diff
@@ -178,6 +178,13 @@ def mail_init_messaging(self, **kwargs):
     def mail_load_message_failures(self, **kwargs):
         return request.env.user.partner_id._message_fetch_failed()
 
+    @http.route('/mail/get_model_definitions', methods=['POST'], type='http', auth='user')
+    def get_model_definitions(self, model_names_to_fetch, **kwargs):
+        return request.make_response(json.dumps(
+            request.env['ir.model']._get_model_definitions(json.loads(model_names_to_fetch)),
+            default=lambda _: False,
```

Comment:
Could we explicitly prevent whatever is making an issue from reaching the dict instead?